libraries/Adapters/Microsoft.Bot.Builder.Adapters.WeChat/Schema/WeChatMessageFactory.cs

```diff
@@ -0,0 +1,198 @@
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardReviewSuccessfulEvent>(doc);
+                                    break;
+                                case "CARD_NOT_PASS_CHECK":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardReviewFailedEvent>(doc);
+                                    break;
+                                case "USER_GET_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardCollectedEvent>(doc);
+                                    break;
+                                case "USER_DEL_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardDeletedEvent>(doc);
+                                    break;
+                                case "USER_GIFTING_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardGiftingEvent>(doc);
+                                    break;
+                                case "USER_CONSUME_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<RemoveAfterUseEvent>(doc);
+                                    break;
+                                case "USER_VIEW_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<ViewCardEvent>(doc);
+                                    break;
+                                case "UPDATE_MEMBER_CARD":
+                                    requestMessage = EntityHelper.FillEntityWithXml<MemberCardUpdatedEvent>(doc);
+                                    break;
+                                case "CARD_SKU_REMIND":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardLowInStockEvent>(doc);
+                                    break;
+                                case "CARD_PAY_ORDER":
+                                    requestMessage = EntityHelper.FillEntityWithXml<CardPointChangeEvent>(doc);
+                                    break;
+                                case "SUBMIT_MEMBERCARD_USER_INFO":
+                                    requestMessage = EntityHelper.FillEntityWithXml<MemberShipActivatedEvent>(doc);
+                                    break;
+                                */
+
+                                // TODO: kf, verify, WeApp, Wxa
+                        }
+
+                        break;
+
+                    default:
+                        {
+                            requestMessage = new UnknowRequest()
+                            {
+                                Content = doc,
+                            };
+                            break;
+                        }
+                }
+            }
+            catch (ArgumentException ex)
+            {
+                logger.LogError(ex, string.Format("RequestMessage Error, MsgType may not exist, XML：{0}", doc.ToString()));
```

Comment:
Pass CultureInfo.InvariantCulture to string.format and remove unnecesary ToString call